### ISSUE TITLE
GraphQL code intel doesn't work

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#9756]
I have the [GraphQL extension](https://sourcegraph.com/extensions/sourcegraph/graphql) enabled, and with https://github.com/sourcegraph/sourcegraph/pull/8099 and https://github.com/sourcegraph/code-intel-extensions/pull/206 my understanding is that GraphQL schema files like https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/schema.graphql#L541:12 (latest master) should have code intel.

But there are no symbols found in the symbols sidebar, and no hovers for types from the schema - instead, I do get hovers that point to same-name symbols found in Go code:

![image](https://user-images.githubusercontent.com/10532611/79043015-afd26e00-7bfc-11ea-9a0a-11d830a6b4e6.png)

The symbols query comes back empty:

![image](https://user-images.githubusercontent.com/10532611/79043084-35561e00-7bfd-11ea-81f4-73b53ffd6e5e.png)

![image](https://user-images.githubusercontent.com/10532611/79043092-3c7d2c00-7bfd-11ea-942d-e7e8e35fec6e.png)

It seems like then the codeintel extension falls back to searching the whole repo, but it doesn't filter it to only GraphQL files (I could see why this _may_ be desirable, but am curious if this is as-designed).

Now, what's weird is that on _some_ commits, I do get symbols and also same-file hovers:

![image](https://user-images.githubusercontent.com/10532611/79043071-1eafc700-7bfd-11ea-9f41-d7ddb647b043.png)

https://sourcegraph.com/github.com/sourcegraph/sourcegraph@133f9631a97f991328048f455f69b62cb38a9c16/-/blob/cmd/frontend/graphqlbackend/schema.graphql#L541:12

Also, the extension page of the GraphQL page seems to show screenshots of Python code?

**Comment by felixfbecker** [comment]
One difference I noticed: When symbols do show up, the GraphQL query had the commit hash passed as the `rev`. But in the request above where it did't work (whenever I'm on latest master with no explicit rev in the URL), `rev` is an empty string, and no symbols are returned by the backend. Maybe the latest revision uses indexed search, and explicit revs (even if they match latest master) use non-indexed search, and indexed symbol search seems to be broken (on files at least).

**Comment by efritz** [comment]
Do you see similar behaviors for non-GraphQL files?

**Comment by felixfbecker** [comment]
I don't really see basic code intel in action a lot since TS and Go have precise code intel, so I unfortunately can't confirm if other languages are affected, but not unlikely that this affects all languages seeing that the symbols sidebar is affected as well

**Comment by felixfbecker** [comment]
FYI I fixed syntax highlighting in #10043

**Comment by efritz** [comment]
@felixfbecker can you confirm this is still an issue? I couldn't reproduce recently.

**Comment by felixfbecker** [comment]
I can reproduce it with the link in the issue description:

![image](https://user-images.githubusercontent.com/10532611/80514311-9a8d6b80-8980-11ea-8b2e-5d15cdacff40.png)

**Comment by efritz** [comment]
I don't know what I was doing earlier. Description in the issue is great.

It looks like there are no symbols for GraphQL if the commit is indexed. If I comment out the call to `searchZoektSymbols` in the git commit resolver this pane populates. I'm not sure how ctags and zoekt interoperate.

I think @keegancsmith may have some additional context to add here.

**Comment by keegancsmith** [comment]
This is likely due to zoekt and symbols service becoming out of sync on the latest version of ctags to use + the .ctags.d. There was a recent bug report about this that @dadlerj was gonna try reproduce on sourcegraph.com https://sourcegraph.slack.com/archives/CHEKCRWKV/p1588056641007800

I'm not 100% convinced this is the root cause, but it's so easy to update and push to sourcegraph.com that I'll do it this morning. That should validate if this is indeed the cause.

There is the larger issue of preventing these getting out of sync. We should probably have a published universal-ctags image with our .ctags.d embedded / embed or .ctags.d into the binary we build.

**Comment by keegancsmith** [comment]
Root cause is a very old regression from when we changed zoekt to use the Sourcegraph user instead of root. This points to us really needing to embed these changes into the universal-ctags binary (we build the binary, so we s hould be able to apply them as patches).

I've fixed it in https://github.com/sourcegraph/zoekt/commit/02b77c3a7a46fda783fb39250ef368a72ea89ade and will shepherd the change into production. Once verified fix I'll close.
